packages/route-plugin/src/plugin.ts

```diff
@@ -36,11 +41,23 @@ plugin.prepareActionCtx(({ ctx, universe }, config: Config) =>
   prepareContext(ctx, config.route.history, universe),
 );
 
-plugin.prepareViewCtx(({ ctx, universe }, config: Config) =>
-  prepareContext(ctx, config.route.history, universe),
-);
+let isSubscribedToHistory = false;
+
+plugin.prepareViewCtx(({ ctx, universe }, config: Config) => {
+  if (!isSubscribedToHistory) {
+    isSubscribedToHistory = true;
+    config.route.history.listen(location =>
+      ctx.dispatch(setRoute)({
+        path: location.pathname,
+        params: parseParams(location.search),
+      }),
+    );
+  }
```

Comment:
Ideally this should be in the `init` method, but we need a `dispatch` function from an action context.